pkg/ccl/changefeedccl/show_changefeed_jobs_test.go

```diff
@@ -116,6 +127,198 @@ func TestShowChangefeedJobsBasic(t *testing.T) {
+				// machine would not work.
+				r.Span.EndKey = tableSpan.Key.Next()
+				return false, nil
+			}
+			if haveGaps {
+				return rnd.Intn(10) > 7, nil
+			}
+			haveGaps = true
+			return true, nil
+		}
+
+		// Checkpoint progress frequently, and set the checkpoint size limit.
+		changefeedbase.FrontierCheckpointFrequency.Override(
+			context.Background(), &s.Server.ClusterSettings().SV, 1)
+		changefeedbase.FrontierCheckpointMaxBytes.Override(
+			context.Background(), &s.Server.ClusterSettings().SV, maxCheckpointSize)
+
+		registry := s.Server.JobRegistry().(*jobs.Registry)
+		foo := feed(t, f, `CREATE CHANGEFEED FOR foo WITH resolved='100ms'`)
+		// Some test feeds (kafka) are not buffered, so we have to consume messages.
+		var shouldDrain int32 = 1
+		g := ctxgroup.WithContext(context.Background())
+		g.Go(func() error {
+			for {
+				if shouldDrain == 0 {
+					return nil
+				}
+				m, err := foo.Next()
+				if err != nil {
+					return err
+				}
+
+				if m.Resolved != nil {
+					ts := extractResolvedTimestamp(t, m)
+					if ts.IsEmpty() {
+						return errors.New("unexpected epoch resolved event")
+					}
+				}
+			}
+		})
+
+		defer func() {
+			closeFeed(t, foo)
+		}()
+
+		jobFeed := foo.(cdctest.EnterpriseTestFeed)
+		loadProgress := func() jobspb.Progress {
+			jobID := jobFeed.JobID()
+			job, err := registry.LoadJob(context.Background(), jobID)
+			require.NoError(t, err)
+			return job.Progress()
+		}
+
+		// Wait for non-nil checkpoint.
+		testutils.SucceedsSoon(t, func() error {
+			progress := loadProgress()
+			if p := progress.GetChangefeed(); p != nil && p.Checkpoint != nil && len(p.Checkpoint.Spans) > 0 {
+				return nil
+			}
+			return errors.New("waiting for checkpoint")
+		})
+
+		// Pause the job and read and verify the latest checkpoint information.
+		require.NoError(t, jobFeed.Pause())
+		progress := loadProgress()
+		require.NotNil(t, progress.GetChangefeed())
+		h := progress.GetHighWater()
+		noHighWater := h == nil || h.IsEmpty()
+		require.True(t, noHighWater)
+
+		jobCheckpoint := progress.GetChangefeed().Checkpoint
+		require.Less(t, 0, len(jobCheckpoint.Spans))
+		var checkpoint roachpb.SpanGroup
+		checkpoint.Add(jobCheckpoint.Spans...)
+
+		// Collect spans we attempt to resolve after when we resume.
+		var resolved []roachpb.Span
+		knobs.FilterSpanWithMutation = func(r *jobspb.ResolvedSpan) (bool, error) {
+			if !r.Span.Equal(tableSpan) {
+				resolved = append(resolved, r.Span)
+			}
+			return false, nil
+		}
+
+		// Resume job.
+		require.NoError(t, jobFeed.Resume())
+
+		// Wait for the high water mark to be non-zero.
+		testutils.SucceedsSoon(t, func() error {
+			prog := loadProgress()
+			if p := prog.GetHighWater(); p != nil && !p.IsEmpty() {
+				return nil
+			}
+			return errors.New("waiting for highwater")
+		})
+
+		var hwt float32
+		var rhwt time.Time
+		sqlDB.QueryRow(t, "SELECT high_water_timestamp, readable_high_water_timestamp from [SHOW CHANGEFEED JOBS]").Scan(&hwt, &rhwt)
```

Comment:
This is the added logic to test this functionality. 
The requires seem unnecessary since if the columns are null in the table, the Scan fails. 
I'm trying to add a way to convert the hwt and rhwt (nit: planning to spell these out) into a type where they can be compared to progress.GetHighWater()